tgui/packages/tgui/interfaces/JobSelection.tsx

```diff
@@ -2,42 +2,47 @@ import { useBackend } from '../backend';
 import { Button, DepartmentEntry, DepartmentPane, Icon, Stack } from '../components';
 import { Window } from '../layouts';
 import { Color } from 'common/color';
+import { merge } from 'common/merge';
 import { SFC } from 'inferno';
 import { JobToIcon } from './common/JobToIcon';
 import { BaseDepartmentInfo, BaseJobInfo } from '../components/DepartmentPane';
 
 class Job implements BaseJobInfo {
-  name: string;
   unavailable_reason: string | null;
   command: boolean;
   open_slots: number;
   used_slots: number;
   icon: string;
   prioritized: boolean;
-  job_description: string;
+  description: string;
 }
 
 class Department implements BaseDepartmentInfo {
-  name: string;
   color: string;
   open_slots: number;
-  jobs: Job[];
+  jobs: { [x: string]: Job };
 }
```

Comment:
These don't have major reasons to be classes. You never instantiate them, and they likely pollute compiled javascript.

The only reason is that you want to check if they implement `BaseDepartmentInfo` interface. But do you actually need to? Functions that expect a `BaseDepartmentInfo` type actually compare fields of the type that you put in to make sure it matches, not whether they implement that specific `BaseDepartmentInfo` symbol. If you pass a type to a function that has incorrect fields, it will show an error with an exact field that is wrong.

So in my humble opinion, you don't really need `class` and `implements`.

You can extend the type instead and save yourself a couple of field duplications, like `color`.

```tsx
type Department = BaseDepartmentInfo & {
  open_slots: number;
  jobs: Record<string, Job>;
};
```

Reply:
Already changed!